### PR TITLE
[TS] Implement Loading and Error Pages for Data Fetching States

### PR DESCRIPTION
**Pull request by a-shigemichi**
## Issue Number
https://github.com/okuda-seminar/Twitter-Clone/issues/472

## Implementation Summary
Added error.tsx file, which follows the official Next.js code conventions. Additionally, removed error rendering and useEffect from my-profile.tsx.

## Particular points to check
Please verify that `error.tsx` is rendered when fetch errors occur on the /profile route by testing different scenarios such as changing the endpoint or username.

## Reference

https://github.com/user-attachments/assets/7e4856d9-f4d8-4e23-b5d8-4736305d0a65

## Test
I was unable to implement manual mocks properly, and therefore could not write the tests scheduled for this week.

## Schedule
2024/11/22

### CODE REVIEW

**Review by TakayaShirai — CHANGES_REQUESTED** [review]
@ryuju0911
In addition to the requested changes, I found that `React.FC` is not compatible with async components when I discussed with ChatGPT. This seems tricky for me to review properly, so I'd appreciate it if you could review this part after I finish my review.

File: ts/twitter/src/app/error.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { useEffect } from "react";
+import { Box, Button } from "@chakra-ui/react";
+
+export default function Error({
```
[the remark this answers is not in the document]
Comment (a-shigemichi) [reply ts/twitter/src/app/error.tsx:6]: error.tsx is a component that gets called when an error is thrown and is never imported directly. I believe using export default is appropriate, similar to page.tsx.

**Review by TakayaShirai — APPROVED** [review]
LGTM. However, please ensure that outdated conversations are resolved once the new commit is pushed

**Review by ryuju0911 — APPROVED** [review]
LGTM

**Comment by ryuju0911** [comment]
@TakayaShirai 
`React.FC` now works with React Server Components and async components. Perhaps ChatGPT provided outdated information.

**Comment by ryuji0123** [comment]
Why not checking official doc or any articles? I've never seen referring to LLM as a primary resource.

**Comment by ryuji0123** [comment]
In case React.FC is incompatible with async components, shouldn't this commit crash the app? I'm not sure why @TakayaShirai pointed out...

File: ts/twitter/src/app/error.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  error,
+  reset,
+}: {
+  error: Error & { digest?: string };
+  reset: () => void;
+}) {
+  useEffect(() => {
+    console.error(error);
```
Comment (ryuji0123) [inline ts/twitter/src/app/error.tsx:14]: BTW, it's great if we can accumulate error in logging platform to evaluate the reliability of our app. Let's talk more on Saturday.

**Review by ryuji0123 — APPROVED** [review]

File: ts/twitter/src/app/error.tsx (hunk 2 above)
Comment (ryuji0123) [reply ts/twitter/src/app/error.tsx:14]: @a-shigemichi 

You can mark this thread resolved once you've checked. In that case, it's common to comment "ACK".

**Comment by TakayaShirai** [comment]
I'm sorry for causing any confusion with this avoidable mistake. I should have noticed that crash would occur if my statement were correct and that it is compatible with async components earlier, even though I reviewed the documentation. I'll make sure to look into details more carefully moving forward. Thank you for pointing this out.

File: ts/twitter/src/app/error.tsx (hunk 2 above)
Comment (a-shigemichi) [reply ts/twitter/src/app/error.tsx:14]: ACK

**Pull request merged**